### PR TITLE
feat(core): Add v3-signing support

### PR DESCRIPTION
**Pull request by Caij**
https://github.com/shwenzhang/AndResGuard/issues/380

### CODE REVIEW

**Review by simpleton — APPROVED** [review]
lgtm

File: AndResGuard-core/src/main/java/com/tencent/mm/androlib/ResourceApkBuilder.java [hunk 1]
```diff
@@ -203,17 +206,17 @@ private void signApkV1(File unSignedApk, File signedApk) throws IOException, Int
...
-      signWithV2sign(unSignedApk, signedApk, minSDKVersion);
+      signWithV2sign(unSignedApk, signedApk, minSDKVersion, signatureType);
       if (!signedApk.exists()) {
         throw new IOException("Can't Generate signed APK v2. Plz check your v2sign info is correct.");
       }
     }
   }
 
-  private void signWithV2sign(File unSignedApk, File signedApk, int minSDKVersion) throws Exception {
+  private void signWithV2sign(File unSignedApk, File signedApk, int minSDKVersion, InputParam.SignatureType signatureType) throws Exception {
```
Comment (simpleton) [inline AndResGuard-core/src/main/java/com/tencent/mm/androlib/ResourceApkBuilder.java:219]: change this function name as well?

**Review by simpleton — COMMENTED** [review]
Thanks for PR, I haven't follow new signature for a while. I have a qq, which part actually use v3 signature? is this `--v3-signing-enabled`?

**Comment by simpleton** [comment]
nvm, found this.
image

**Comment by zhangwanggit** [comment]
When to update to 1.2.21

**Pull request merged**